demo/src/main.tsx

```diff
@@ -3,17 +3,39 @@ import ReactDOM from 'react-dom/client';
 import App from './App';
 import './index.css';
 
-import { SpaceContextProvider } from './components';
 import { SlidesStateContextProvider } from './components/SlidesStateContext.tsx';
+import Spaces from '@ably/spaces';
+import { SpacesProvider, SpaceProvider } from '@ably/spaces/react';
+import { Realtime } from 'ably';
+import { nanoid } from 'nanoid';
+import { generateSpaceName, getParamValueFromUrl } from './utils';
+import { AblyProvider } from 'ably/react';
 
 const root = ReactDOM.createRoot(document.getElementById('root') as HTMLElement);
 
+const clientId = nanoid();
+
+const client = new Realtime.Promise({
+  authUrl: `/api/ably-token-request?clientId=${clientId}`,
+  clientId,
+});
+
+const spaces = new Spaces(client);
+const spaceName = getParamValueFromUrl('space', generateSpaceName);
+
 root.render(
   <React.StrictMode>
-    <SpaceContextProvider>
-      <SlidesStateContextProvider>
-        <App />
-      </SlidesStateContextProvider>
-    </SpaceContextProvider>
+    <SpacesProvider client={spaces}>
+      <AblyProvider client={client}>
```

Comment:
Should the `AblyProvider` not be the top provider, even if just for conceptual reasons (i.e. Ably client being a dependency of Spaces)?

Reply:
Agree, fixed